### ISSUE TITLE
SIGSEGV in setAnchorPosition

### ISSUE THREAD

**Issue by agriggio** [artpixls/ART#292]
**[Original report](https://bitbucket.org/agriggio/art/issues/292) by Olivier Croquette (Bitbucket: [Olivier Croquette](https://bitbucket.org/Olivier Croquette), ).**

----------------------------------------

I have updated recently to 1.19.3 and I am experiencing since then random crashes. I was able to catch one in gdb. Here is the stack trace.

```
Thread 1 "ART.bin" received signal SIGSEGV, Segmentation fault.
0x000055555576a439 in CropHandler::setAnchorPosition(int, int, bool) ()
(gdb) bt
#0  0x000055555576a439 in CropHandler::setAnchorPosition(int, int, bool) ()
#1  0x000055555576f2aa in CropWindow::zoomFit() ()
#2  0x0000555555770cca in CropWindow::updateCursor(int, int, int) ()
#3  0x0000555555777731 in CropWindow::pointerMoved(int, int, int, double) ()
#4  0x00005555558a12fb in ImageArea::on_motion_notify_event(_GdkEventMotion*) ()
#5  0x00007ffff1bc7ca4 in Gtk::Widget_Class::motion_notify_event_callback(_GtkWidget*, _GdkEventMotion*) ()
   from /tmp/tmp.pWaaiQaDsK-ART/ART/lib/libgtkmm-3.0.so.1
#6  0x00007ffff6c2fba1 in ?? () from /tmp/tmp.pWaaiQaDsK-ART/ART/lib/libgtk-3.so.0
#7  0x00007ffff46101a4 in ?? () from /tmp/tmp.pWaaiQaDsK-ART/ART/lib/libgobject-2.0.so.0
#8  0x00007ffff462a3a1 in g_signal_emit_valist () from /tmp/tmp.pWaaiQaDsK-ART/ART/lib/libgobject-2.0.so.0
#9  0x00007ffff462afbf in g_signal_emit () from /tmp/tmp.pWaaiQaDsK-ART/ART/lib/libgobject-2.0.so.0
#10 0x00007ffff6d7c57c in ?? () from /tmp/tmp.pWaaiQaDsK-ART/ART/lib/libgtk-3.so.0
#11 0x00007ffff6c2cb2e in ?? () from /tmp/tmp.pWaaiQaDsK-ART/ART/lib/libgtk-3.so.0
#12 0x00007ffff6c2ebee in gtk_main_do_event () from /tmp/tmp.pWaaiQaDsK-ART/ART/lib/libgtk-3.so.0
#13 0x00007ffff6636e15 in ?? () from /tmp/tmp.pWaaiQaDsK-ART/ART/lib/libgdk-3.so.0
#14 0x00007ffff6667ea2 in ?? () from /tmp/tmp.pWaaiQaDsK-ART/ART/lib/libgdk-3.so.0
#15 0x00007ffff424a7f7 in g_main_context_dispatch () from /tmp/tmp.pWaaiQaDsK-ART/ART/lib/libglib-2.0.so.0
#16 0x00007ffff424aa60 in ?? () from /tmp/tmp.pWaaiQaDsK-ART/ART/lib/libglib-2.0.so.0
#17 0x00007ffff424ab0c in g_main_context_iteration () from /tmp/tmp.pWaaiQaDsK-ART/ART/lib/libglib-2.0.so.0
#18 0x00007ffff4a9f7bd in g_application_run () from /tmp/tmp.pWaaiQaDsK-ART/ART/lib/libgio-2.0.so.0
#19 0x00005555556f4f44 in main ()
```

‌

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

Hi,

Is this from the linux64 binary, or did you build art yourself? If the latter, can you try with a debug build and give me more info about your environment?

**Comment by agriggio** [comment]
**Original comment by Olivier Croquette (Bitbucket: [Olivier Croquette](https://bitbucket.org/Olivier Croquette), ).**

----------------------------------------

Hi, it is from the official linux64 binary. I can reproduce it now, it happens when I have started an interactive action with the cropping tool and I move to the next file with F4

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

Thanks, I'll try to reproduce and go from there then!

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

I didn’t manage to reproduce. Would you be able to create a short video that shows the steps you are taking? Thanks!

**Comment by agriggio** [comment]
**Original comment by Olivier Croquette (Bitbucket: [Olivier Croquette](https://bitbucket.org/Olivier Croquette), ).**

----------------------------------------

I cannot reproduce it anymore either. Maybe there are prerequisites to trigger the crash. I will keep you updated if I find out more.

**Comment by agriggio** [comment]
**Original comment by apostel338 (Bitbucket: [apostel338](https://bitbucket.org/apostel338), GitHub: [apostel338](https://github.com/apostel338)).**

----------------------------------------

You can see the moment when I hit F4 in the windows title (title changes to next photo). ART hangs at that moment and takes a while to crash.

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

prevent segfault when switching images while editing the crop

Fixes #292

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

Thanks, I managed to reproduce it now. I just pushed a patch that should fix this hopefully
